Sources/MapboxCoreNavigation/HistoryFileReader.swift

```diff
@@ -0,0 +1,148 @@
+import Foundation
+import MapboxNavigationNative
+
+/// Digest of history file contents produced by `HistoryFileReader`.
+public struct HistoryFileDump {
+    /// Initial `IndexedRouteResponse` that was set to the Navigator.
+    ///
+    /// Can be `nil` if current file is a free drive recording or if history recording was started after such event. In latter case this property may contain another `IndexedRouteResponse` which was, for example, set as a result of a reroute event.
+    public fileprivate(set) var initialRoute: IndexedRouteResponse?
+    /// Stream of location updates.
+    public fileprivate(set) var rawLocations: [CLLocation] = []
+}
+
+/// Provides event-by-event access to history files contents.
```

Comment:
Either this class comment or one of the method comments below should mention that `HistoryRecording.stopRecordingHistory(writingFileWith:)` writes a file to `HistoryRecording.historyDirectoryURL` that this class can read.